### ISSUE TITLE
PHP8.2 Support in CMS5 for core modules

### ISSUE THREAD

**Issue by maxime-rainville** [silverstripe/silverstripe-framework#10403]
The [first alpha of PHP8.2](https://php.watch/news/2022/06/php82-alpha1) is available. It's meant to reach feature freeze on July 19.
  
### Acceptance criteria
- [x] ~~We have a clearly documented way of running a dev environment on pre-release PHP8.2. That's mostly targeted toward the internal SIlverstripe dev team.~~ You should be able to install PHP 8.2 just like any other version as it's now stable
- [ ] CMS5 core ~~and supported module~~ builds can run PHP8.2. **<< I've split off a [new issue for supported modules]**(https://github.com/silverstripe/silverstripe-framework/issues/10618)
- [x] [Syntax/constructs deprecated in PHP8.2](https://php.watch/versions/8.2) are no longer used.

### Notes
- Queued Job will probably hate the [Dynamic Properties are deprecated](https://php.watch/versions/8.2/dynamic-properties-deprecated), If it requires substantial refactor, split it off as a separate card.
- https://github.com/symbiote/silverstripe-queuedjobs/issues/377

### New issues created
- https://github.com/silverstripe/silverstripe-graphql/issues/511 << Blocks PHP 8.2 support for us
- https://github.com/silverstripe/silverstripe-framework/issues/10618

### Third party issues created
- https://github.com/symfony/symfony/issues/48637 **<< this may be a non-issue for us, I didn't see this warning during testing, however I did detect it doing a string search for dynamic properties for things that could potentially go wrong e.g. not covered by unit tests**
 
### Third party PRs
- https://github.com/webonyx/graphql-php/pull/1256 **<< PR was for 14.x - only 15 (in beta) is supported, have asked when stable for 15 is being released - for our policy we should technically be using a stable not a beta**
- https://github.com/oscarotero/html-parser/pull/4

### CI Integration test
**- https://github.com/emteknetnz/silverstripe-installer/actions/runs/3711071825**

### PRs
- ~~https://github.com/silverstripe/gha-generate-matrix/pull/39~~ **<< Merge on the non-core PHP 8.2 card**
- https://github.com/silverstripe/silverstripe-framework/pull/10614
- https://github.com/silverstripe/silverstripe-assets/pull/535
- https://github.com/silverstripe/silverstripe-versioned/pull/386
- https://github.com/silverstripe/silverstripe-assets/pull/533
- https://github.com/silverstripe/silverstripe-config/pull/77
- https://github.com/silverstripe/silverstripe-framework/pull/10628

**Comment by emteknetnz** [comment]
We're blocked on:
- webonyx/graphql-php v15 being tagged stable / us upgrading silverstripe/graphl to use v15
- ~~oscarotero/html-parser PR being merged and tagged~~
- CMS 5 compatible kitchen-sink branches being available

I've reduced the scope of this issue to "core" modules only and I've created a new issue to handle kitchen-sink

Given that graphql is "core", even when we close this issue we still won't have core support for PHP 8.2 - though we should once we've completed the [webonyx/graphql-php v15 issue](https://github.com/silverstripe/silverstripe-graphql/issues/511)

**Comment by GuySartorelli** [comment]
@emteknetnz Your external PR https://github.com/oscarotero/html-parser/pull/4 has been merged and tagged - do we need to change the minimum depencency or are we okay with `--prefer-lowest` not being 8.2 compatible? I think just leave as is but wanna confirm.

If we don't need to change that, this can be closed.

**Comment by emteknetnz** [comment]
May as well update the minimum installable version https://github.com/silverstripe/silverstripe-framework/pull/10628
